File: backend/team6/src/test/java/programmers/team6/domain/member/service/DeptServiceTest.java
package programmers.team6.domain.member.service;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import programmers.team6.domain.member.repository.DeptRepository;
import programmers.team6.global.exception.code.NotFoundErrorCode;
import programmers.team6.global.exception.customException.NotFoundException;

@ExtendWith(MockitoExtension.class)
class DeptServiceTest {
	@InjectMocks
	DeptService deptService;
	@Mock
	DeptRepository deptRepository;

	@Test
	void should_throwNotFoundException_when_givenNotExistDeptName() {
		// when & then
		Assertions.assertThatThrownBy(() -> deptService.findByDeptName("EMPTY"))
			.isInstanceOf(NotFoundException.class)
			.hasMessage(NotFoundErrorCode.NOT_FOUND_DEPT.getMessage());
	}

}

File: backend/team6/src/test/java/programmers/team6/domain/admin/service/DeptServiceTest.kt
package programmers.team6.domain.admin.service

import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import programmers.team6.domain.admin.repository.DeptRepository
import programmers.team6.global.exception.code.NotFoundErrorCode
import programmers.team6.global.exception.customException.NotFoundException

internal class DeptServiceTest {
    private var deptRepository = mockk<DeptRepository>()

    private var deptService = DeptService(deptRepository)

    @Test
    fun should_throwNotFoundException_when_givenNotExistDeptName() {
        // given
        every { deptRepository.findByDeptName("EMPTY") } returns null

        // when & then
        Assertions.assertThatThrownBy { deptService.findByDeptName("EMPTY") }
            .isInstanceOf(NotFoundException::class.java)
            .hasMessage(NotFoundErrorCode.NOT_FOUND_DEPT.message)
    }
}